tasks/release.py

```diff
@@ -35,6 +35,117 @@ def add_prelude(ctx, version):
     ctx.run("git commit -m \"Add prelude for {} release\"".format(version))
 
 
+@task
+def add_dca_prelude(ctx, version, agent7_version, agent6_version=""):
+    """
+    Release of the Cluster Agent should be pinned to a version of the Agent.
+    """
+    res = ctx.run("reno --rel-notes-dir releasenotes-dca new prelude-release-{0}".format(version))
+    new_releasenote = res.stdout.split(' ')[-1].strip()  # get the new releasenote file path
+
+    if agent6_version != "":
+        version_split = agent6_version.split(".")
+        agent6_version = "--6{}{}".format(
+            version_split[1], version_split[2]
+        )  # generate the right hyperlink to the agent's changelog.
+
+    with open(new_releasenote, "w") as f:
+        f.write(
+            """prelude:
+    |
+    Released on: {1}
+    Pinned to datadog-agent v{0}: `CHANGELOG <https://github.com/DataDog/datadog-agent/blob/master/CHANGELOG.rst#{2}{3}>`_.""".format(
+                agent7_version, date.today(), agent7_version.replace('.', ''), agent6_version,
+            )
+        )
+
+    ctx.run("git add {}".format(new_releasenote))
+    ctx.run("git commit -m \"Add prelude for {} release\"".format(version))
+
+
+@task
+def update_dca_changelog(ctx, new_version, agent_version):
+    """
+    Quick task to generate the new CHANGELOG-DCA using reno when releasing a minor
+    version (linux/macOS only).
+    """
+    new_version_int = list(map(int, new_version.split(".")))
+
+    if len(new_version_int) != 3:
+        print("Error: invalid version: {}".format(new_version_int))
+        raise Exit(1)
+
+    agent_version_int = list(map(int, agent_version.split(".")))
+
+    if len(agent_version_int) != 3:
+        print("Error: invalid version: {}".format(agent_version_int))
+        raise Exit(1)
+
+    # let's avoid losing uncommitted change with 'git reset --hard'
+    try:
+        ctx.run("git diff --exit-code HEAD", hide="both")
+    except Failure:
+        print("Error: You have uncommitted change, please commit or stash before using update_changelog")
+        return
+
+    # make sure we are up to date
+    ctx.run("git fetch")
+
+    # let's check that the tag for the new version is present (needed by reno)
+    try:
+        ctx.run("git tag --list | grep {}".format(new_version))
+    except Failure:
+        print("Missing '{}' git tag: mandatory to use 'reno'".format(new_version))
+        raise
+
+    # removing releasenotes from bugfix on the old minor.
+    # cluster agent minor releases are in sync with the agent's, bugfixes are not necessarily.
+    branching_point_agent = "{}.{}.0-devel".format(agent_version_int[0], agent_version_int[1])
+    previous_minor_branchoff = "dca-{}.{}.X".format(new_version_int[0], new_version_int[1] - 1)
```

Comment:
I think this would benefit from some more explanations, either here or in the dca release docs. it's not immediately clear how this work. Giving explicit assumptions and an example may make this clearer, eg.:
- The script assumes that the dca bugfix branches diverge from master on the commit where the corresponding Agent `devel` tag is.
- `dca-1.9.0` is based on Agent `7.23.0`, thus `agent_version=7.23.0` and `new_version=1.9.0`.
- `dca-1.8.0` is based on Agent `7.22.0`. Based on the above assumption, the `dca-1.8.X` branch diverges from master on the commit where `7.23.0-devel` is tagged. 
- Therefore, to avoid including release notes that were added in a `1.8.x` bugfix, removing all files added between `7.23.0-devel` (= `branching_point_agent `) and `dca-1.8.X` (= `previous_minor_branchoff`)  is the correct thing to do.


--------

An alternative solution (to not implicitly depend on Agent versions being correctly tagged) would be to have dedicated `dca-1.X.0-devel` tags. That tag would simplify the logic here, as well as allow you to use `--earliest-version` in the below reno command to not search the whole tree for release notes.

But you may want to keep the current scheme to make sure that DCA releases are correctly sync'd with Agent releases.

Reply:
Added a note here, but I am using your example in our release documentation.